components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -79,6 +83,41 @@
         return groups;
     }
 
+    /**
+     * Lists the groups that belong to a tenant by its tenant id that are created from SCIM.
+     *
+     * @param tenantId tenant Id
+     * @return The set of groups that were created from SCIM
+     * @throws IdentitySCIMException If an error occurred while reading from persistence store.
+     */
+    public Set<String> listSCIMGroups(int tenantId) throws IdentitySCIMException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
+        PreparedStatement prepStmt = null;
+        ResultSet resultSet = null;
+        Set<String> groups = new HashSet<>();
+
+        try {
+            //Retrieve groups from the DB.
+            prepStmt = connection.prepareStatement(SQLQueries.LIST_SCIM_GROUPS_BY_TENANT_ID_SQL);
+            prepStmt.setInt(1, tenantId);
+            prepStmt.setString(2, SCIMConstants.CommonSchemaConstants.ID_URI);
+            resultSet = prepStmt.executeQuery();
+            while (resultSet.next()) {
+                String group = resultSet.getString(1);
+                if (StringUtils.isNotEmpty(group)) {
+                    group = SCIMCommonUtils.getPrimaryFreeGroupName(group);
+                    groups.add(group);
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentitySCIMException("Error when reading the SCIM Group information from persistence store.", e);
+        } finally {
```

Comment:
finally block can be removed when the try with resource is used.

Reply:
Removed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/252/commits/734e0608b7a8b0b0f7e299af86bda0d092285945